app/uk/gov/hmrc/securemessage/controllers/models/generic/ApiConversation.scala

```diff
@@ -47,34 +47,29 @@ object ApiConversation {
       messages = conversation.messages.map(message => convertToApiMessage(conversation, message, identifier))
     )
 
+  private def isSender(sender: Option[Participant], reader: Option[Participant]) =
+    for {
+      s <- sender
+      r <- reader
+    } yield (s.id === r.id)
+
   private def convertToApiMessage(
     coreConversation: Conversation,
     message: Message,
     identifier: Identifier): ApiMessage = {
-    val sender: Option[Participant] = findParticipantViaId(coreConversation, message.senderId)
-    val reader: Option[Participant] = findParticipantViaIdentifier(coreConversation, identifier)
-    (sender, reader) match {
-      // you're the sender
-      case (Some(participantSender), Some(participantReader)) if participantSender.id === participantReader.id =>
-        ApiMessage(None, Some(message.created), None, message.content)
-      // not the sender, but am the first reader
-      case (Some(participantSender), Some(participantReader))
-          if isFirstReader(participantReader, message, coreConversation) =>
-        ApiMessage(
-          Some(SenderInformation(participantSender.name, message.created)),
-          None,
-          firstReaderInformation(coreConversation, message),
-          message.content
-        )
-      // not the sender, and not the first reader
-      case (Some(participantSender), Some(_)) =>
-        ApiMessage(
-          Some(SenderInformation(participantSender.name, message.created)),
-          None,
-          firstReaderInformation(coreConversation, message),
-          message.content
-        )
-      case (_, _) => ApiMessage(None, None, None, message.content)
+    val senderPossibly: Option[Participant] = findParticipantViaId(coreConversation, message.senderId)
```

Comment:
guess you can just name it sender, the type Option is clear that this is an optional parameter :)